### PR TITLE
chore(release): Bump to version 0.5.0

### LINKED ISSUE #2381: Feature request - Show possible download size in `status` command
When doing a > scoop list command, it would be very helpful to show what apps have updates available, next to it's version number.  Yes, I know that doing scoop update shows new available, but when having many apps installed it's hard to keep track of all that, and the main update command only shows the most recent anyway, not ones it listed in the past.

Also, a upgrade command, like what apt-get does would help:
> scoop upgrade

much like the way apt-get upgrade works, it would be nice for scoop to get the list of upgradable apps, then have it show which ones will upgrade and the amount of data it may take.

Oh, speaking of, it may be helpful to actually put the file sizes in the scoop buckets and have the install command confirm the download first, showing the file size.  I have a data cap, so every byte counts.

### LINKED ISSUE #4327: cached files with URL and filename > 260 characters issue and possible solution
I'm making a manifest for Microsoft Edge and I'm getting the filename too long problem.

```
The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
URL http://msedge.f.tlu.dl.delivery.mp.microsoft.com/filestreamingservice/files/c6c3d5da-8cd4-4d17-a52d-9c1b4e9f4ff0?P1=1619294442&P2=404&P3=2&P4=W82qhGeOfF%2bacPu5vkB5FNoZKDucQJEnJ87E0P%2bMAYsiVsgr%2fsrFQA%2fnkdaAVPqIh6m%2b83hjYL%2fCtbpRhTCOXg%3d%3d#/dl.7z is not valid
```

I don't see anyway to make the URL shorter as this is the full URL, the problem is caused by the `cache` dir that caches apps with `$app $version $url` 

https://github.com/lukesampson/scoop/blob/3d67b7d37c4a2714e8942724d4677dcbd3678f01/lib/install.ps1#L107

Is there any reason we actually need that URL there ? `$app $version` should be enough I think ?

If not we can:
-  add the hash instead of the url
-  or make our own hash of `$app $version $url`

Also I did enable the `LongPathsEnabled` in the registry and I'm on Windows 10 2004, but this didn't fix the issue.
I'm now running a fork with only `$app $version` in the `cache` directory and it works fine.

Any thoughts?
(I can make a PR for this)

### LINKED ISSUE #5920: [Bug] `json_path` returning string literal `\n` when using union operator
#### Current Behavior
Currently, `json_path` would return the string literal `\n` when using union operator (`[,]`).

#### Expected Behavior
Instead, `json_path` should return the values with new lines, separated by commas, and contained in brackets (`[]`).

I am trying to extract the version and UUID from this JSON <https://api.azul.com/metadata/v1/zulu/packages?java_version=17&os=windows&arch=x86&archive_type=zip&java_package_type=jdk&javafx_bundled=false&crac_supported=false&latest=true&page_size=2>.

Using this JSON path should returns 4 different values
```
$..['distro_version','package_uuid']
```

> Note that <https://jsonpath.com/> only accept this syntax
> ```
> $..distro_version,package_uuid
> ```

Currently, the returned value is
```
[
  17,
  48,
  15,
  0
]\ncef00b50-77db-400a-afdd-b9e391e44631\n[
  17,
  48,
  15,
  0
]\n25351721-5e7e-4eef-a0ef-0c3686b38631
```

However, I don't think this is a feature and trying to the match regex for it would break if this is changed/fixed.

### System details

- **Windows version:** 11
- **OS architecture:** 64bit
- **PowerShell version:** 7.4.2

### LINKED ISSUE #5948: [Bug] `use_sqlite_cache` will store all the manifest files in the bucket folder
#### Current Behavior

如下方截图所示，设置了 `use_sqlite_cache: true` 时，database内会包含对应 `bucket` 文件夹内的所有 `.json` 清单文件，但是除开了桶文件夹中的 `bucket` 文件夹内的清单文件，其他的文件是无效的，导致在搜索之后以为有这个软件清单，但安装时直接报未找到的错误，比如截图中安装的文件在 https://github.com/chawyehsu/dorado/tree/master/deprecated 内

#### Expected Behavior

正确的储存有效的清单文件信息

#### Additional context/output

when set `use_sqlite_cache: true`

image

when set `use_sqlite_cache: false`

image

### System details

**Windows version:** 11

**PowerShell version:** 7.4.2

#### Scoop Configuration

```yaml
scoop_branch: develop
use_sqlite_cache : true
```

### LINKED ISSUE #5952: [Bug] using any package gives error
#### Current Behavior
using any package gives error

scoop install neofetch
Installing 'neofetch' (7.1.0) [64bit] from 'main' bucket
neofetch (333.6 KB) [=========================================================================================] 100%
Checking hash of neofetch ... ok.
Linking ~\scoop\apps\neofetch\current => ~\scoop\apps\neofetch\7.1.0
Creating shim for 'neofetch'.
'neofetch' (7.1.0) was installed successfully!
'neofetch' suggests installing 'git'.

neofetch
/bin/bash: C:\Users\nirgr\scoop\apps\neofetch\current\neofetch: No such file or directory

#### Expected Behavior
work as expected

#### Additional context/output
I have WSL and Cygwin installed

### System details

**Windows version:** Win 11 Version 23H2 (0S Build 22631.3447)

### LINKED ISSUE #5963: [Bug] Removing buckets doesn't remove corresponding sqlite caches
## Bug Report

`scoop bucket rm <bucket>` Removing buckets doesn't clear corresponding sqlite caches

#### Current Behavior

```sh
scoop config use_sqlite_cache true
scoop bucket add versions
scoop bucket rm ersions
scoop search nodejs-nightly
# nodejs-nightly manfiest from the Versions bucket remains available
```

#### Expected Behavior

```
# nodejs-nightly manfiest from the Versions bucket should not be available
```

#### Possible Solution

Perform sqlite cache update when removing a bucket.

### System details

Irrelevant

#### Scoop Configuration

```
{
  ...
  "scoop_branch": "develop",
  "use_sqlite_cache": true
}
```

### LINKED ISSUE #5965: `scoop.db` is read and locked during the whole lifetime of current PowerShell session
## Bug Report

`scoop.db` is locked during the whole lifetime of current PowerShell session

#### Current Behavior

```sh
scoop config use_sqlite_cache true
# `scoop.db` is locked during the whole lifetime of current PowerShell session
```

#### Expected Behavior

`scoop.db` should be closed and released after operations(whether creation, update or deletion) to the cache db.

### System details

Irrelevant

#### Scoop Configuration

```
{
  ...
  "scoop_branch": "develop",
  "use_sqlite_cache": true
}
```

### LINKED ISSUE #5977: Autoupdate to manifest url
Hello everyone!

I was wondering if there was any way i could use the `autoupdate`  name@version feature with a manifest specified as a path and not from a bucket.

For example lets say i have a `manifest.json` that has an `autoupdate` field. If I were to run:

`scoop install ./manifest.json@20`

I would receive the error: `Couldn't find manifest for 'manifestjson@20'`

Is there any workaround to this? Any flags i could set?

### LINKED ISSUE #6012: [Bug] Invoke-Git : The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as the name of a cmdlet, function, script file, or operable  program.
#### Current Behavior

Pretty much just continue to get the error in the title whenever I run a `scoop update` without it doing much else, last time I had ran an update was like ~2ish weeks ago and it was working fine then.

#### Expected Behavior

For it to continue to work like it was before...

#### Additional context/output

the full error:
```
scoop update
Updating Scoop...
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:283
Line |
 283 |          return & $git @ArgumentList
     |                   ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:301
Line |
 301 |      return & $git @ArgumentList
     |               ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:283
Line |
 283 |          return & $git @ArgumentList
     |                   ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:301
Line |
 301 |      return & $git @ArgumentList
     |               ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:283
Line |
 283 |          return & $git @ArgumentList
     |                   ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:301
Line |
 301 |      return & $git @ArgumentList
     |               ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:283
Line |
 283 |          return & $git @ArgumentList
     |                   ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:301
Line |
 301 |      return & $git @ArgumentList
     |               ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:283
Line |
 283 |          return & $git @ArgumentList
     |                   ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:301
Line |
 301 |      return & $git @ArgumentList
     |               ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:283
Line |
 283 |          return & $git @ArgumentList
     |                   ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
ObjectNotFound: The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:283
Line |
 283 |          return & $git @ArgumentList
     |                   ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
ObjectNotFound: The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:283
Line |
 283 |          return & $git @ArgumentList
     |                   ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
ObjectNotFound: The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:283
Line |
 283 |          return & $git @ArgumentList
     |                   ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
&: C:\Users\JourneyOver\scoop\apps\scoop\current\lib\core.ps1:301
Line |
 301 |      return & $git @ArgumentList
     |               ~~~~
     | The term 'c:\program files\git\cmd\git.exe c:\program files\git\mingw64\bin\git.exe' is not recognized as a name of a
     | cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
```

### System details

**Windows version:** 
Windows 11

**OS architecture:** 
64bit

**PowerShell version:**

Major  Minor  Build  Revision
5      1      22621  2506 

**Additional software:**

Git
Cmder (Have tested on terminal, command prompt and powershell and all of them produce the same error)

#### Scoop Configuration

```json
{
  "last_update": "2024-06-12T18:38:50.4890679-05:00",
  "scoop_branch": "develop",
  "scoop_repo": "https://github.com/ScoopInstaller/Scoop"
}
```

### PR DESCRIPTION
**Pull request by niheaven**
## ✨ Highlights

- Scoop has implemented SQLite for caching app information, which accelerates local searches and the installation of older app versions. Enable this feature by using the command `scoop config use_sqlite_cache true`.
- The cache names for Scoop apps are derived from the first 7 bytes of the SHA256 hash of the download URL, which helps to shorten the file names.

### Features

- **scoop-search:** Use SQLite for caching apps to speed up local search (#5851, #5918, #5946, #5949, #5955, #5966, #5967, #5981)
- **core:** New cache filename format (#5929, #5944)
- **decompress:** Use innounp-unicode as default Inno Setup Unpacker (#6028)
- **install:** Added the ability to install specific version of app from URL/file link (#5988)

### Bug Fixes

- **scoop-download|install|update:** Use consistent options (#5956)
- **scoop-info:** Fix download size estimating (#5958)
- **scoop-search:** Catch error of parsing invalid manifest (#5930)
- **checkver:** Correct variable 'regex' to 'regexp' (#5993)
- **checkver:** Correct error messages (#6024)
- **core:** Search for Git executable instead of any cmdlet (#5998)
- **core:** Use correct path in 'bash' (#6006)
- **core:** Limit the number of commands to get when search for git executable (#6013)
- **decompress:** Match `extract_dir`/`extract_to` and archives (#5983)
- **json:** Serialize jsonpath return (#5921)
- **shim:** Restore original path for JAR cmd (#6030)

### Code Refactoring

- **decompress:** Use 7zip to extract Zstd archive (#5973)
- **install:** Separate archive extraction from downloader (#5951)
- **install:** Replace 'run_(un)installer()' with 'Invoke-Installer()' (#5968, #5971)

## Close Issues

- Closes #5920
- Closes #4327
- Closes #5948
- Closes #2381
- Closes #5965
- Closes #5963
- Closes #5977
- Closes #5952
- Closes https://github.com/ScoopInstaller/Extras/issues/13385
- Closes https://github.com/ScoopInstaller/Extras/pull/13342
- Closes #6012
- Related #6026
- Closes https://github.com/ScoopInstaller/Main/pull/5920 and related issues

## Credits

Thanks to the following contributors (ordered by PR merging time):

@niheaven @chawyehsu @lewis-yeung @MatejKafka @qwertyhjklxyz

### CODE REVIEW

File: lib/core.ps1 [hunk 1]
```diff
@@ -1038,22 +1064,30 @@ function shim($path, $global, $name, $arg) {
...
+            ')'
         ) -join "`r`n" | Out-UTF8File "$shim.cmd"
 
         warn_on_overwrite $shim $path
         @(
-            "#!/bin/sh",
+            '#!/bin/sh',
             "# $resolved_path",
-            "`"$resolved_path`" $arg `"$@`""
+            "if [ `$WSL_INTEROP ]",
```
Comment (rasa) [inline lib/core.ps1:1085]: Shouldn't there be a closing <tt>`</tt> here?

Comment (rasa) [inline lib/core.ps1:1085]:
```suggestion
            "if [ `$WSL_INTEROP` ]",
```
Doesn't a closing <tt>`</tt> belong here?

**Review by rasa — APPROVED** [review]
I haven't tested them, but the changes look great. I will switch to this PR, and if I encounter any issues, I will file them.

File: lib/core.ps1 (hunk 1 above)
Comment (MatejKafka) [reply lib/core.ps1:1085]: The ` is escaping the $.

**Comment by niheaven** [comment]
I will merge it today and check for any bugs based on user feedback.

**Pull request merged**